### PR TITLE
fix(#32): fix starting app on MACos

### PR DESCRIPTION
**Pull request by Derane**
## Description
fix starting app on MACos

## Related Issue

https://github.com/VilnaCRM-Org/user-service/issues/32
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING.md**](https://github.com/VilnaCRM-Org/php-service-template/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] You have only one commit (if not, squash them into one commit).

### CODE REVIEW

File: infrastructure/docker/php/docker-entrypoint.sh [hunk 1]
```diff
@@ -60,9 +60,10 @@ if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
 			bin/console doctrine:migrations:migrate --no-interaction
 		fi
 	fi
-
-	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
-	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
+  if touch var/test_acl_support && setfacl -m u:www-data:rwx var/test_acl_support 2>/dev/null; then
+        setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
+        setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
+        rm -f var/test_acl_support
```
Comment (paaneko) [inline infrastructure/docker/php/docker-entrypoint.sh:66]:
```suggestion
  if [ "$APP_ENV" = 'prod' ]; then
    setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
    setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
  fi
```
OR this. That worked for me, too.

**Comment by paaneko** [comment]
I noticed I forgot to close the `if` tag with `fi`)

File: infrastructure/docker/php/docker-entrypoint.sh [hunk 2]
```diff
@@ -60,9 +60,9 @@ if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
 			bin/console doctrine:migrations:migrate --no-interaction
 		fi
 	fi
-
-	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
-	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
+  if [ "$APP_ENV" = 'prod' ]; then
+    setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
+    setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
```
Comment (paaneko) [inline infrastructure/docker/php/docker-entrypoint.sh:65]:
```suggestion
    setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
  fi
```

**Comment by paaneko** [comment]
Weirdly, on my mac everything runs good.

File: infrastructure/docker/php/docker-entrypoint.sh [hunk 3]
```diff
@@ -60,12 +60,7 @@ if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
 			bin/console doctrine:migrations:migrate --no-interaction
 		fi
 	fi
-  if [ "$APP_ENV" = 'prod' ]; then
-      setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
-      setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
-  else
-      chmod -R 755 var
-  fi
+  chmod -R 755 var
```
Comment (Kravalg) [inline infrastructure/docker/php/docker-entrypoint.sh:63]: Let's use setfacl if it's supported
In other cases we can use chmod

For example:
```
if command -v setfacl >/dev/null 2>&1; then
    # Check if ACLs are supported by trying to apply an ACL
    if setfacl -m u::rwx "$TARGET_DIR" 2>/dev/null; then
        echo "ACLs are supported. Setting ACLs..."
        setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX "$TARGET_DIR"
    else
        echo "ACLs are not supported. Falling back to chmod..."
        chmod 755 -R "$TARGET_DIR"
    fi
else
    echo "setfacl command not found. Using chmod..."
    chmod 755 -R "$TARGET_DIR"
fi
```

**Comment by paaneko** [comment]
2024-07-18 20:04:22 Waiting for db to be ready...
2024-07-18 20:04:22 The db is now ready and reachable
2024-07-18 20:04:22 
2024-07-18 20:04:22  [OK] Already at the latest version ("DoctrineMigrations\Version20240122085853")
2024-07-18 20:04:22 
2024-07-18 20:04:22 ACLs are supported. Setting ACLs...
2024-07-18 20:04:22 setfacl: var: Not supported

That my docker container logs.

File: .github/workflows/E2Etests-mac.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: E2E testing-mac
+
+on:
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  behat:
+    name: Behat
+    runs-on: macos-13
```
Comment (Kravalg) [inline .github/workflows/E2Etests-mac.yml:10]: Let’s try to use macOS 14

**Review by Kravalg — CHANGES_REQUESTED** [review]

**Review by Kravalg — CHANGES_REQUESTED** [review]
Let’s fix the conflicts and then we will be able to merge this PR

**Review by Kravalg — APPROVED** [review]

**Pull request merged**